tmt/steps/discover/fmf.py

```diff
@@ -237,15 +241,15 @@ def go(self):
                 "Cannot manipulate with dist-git without "
                 "the `--dist-git-merge` option.")
 
-        def get_git_root(dir):
-            return self.run(
+        def get_git_root(dir: str) -> str:
+            return cast(str, self.run(
```

Comment:
This `cast()` is suspicious. `self.run()` returns `tmt.utils.CommandOutput`, and its `.stdout` attribute is `Optional[str]`. So `str` is already there, but it may be `Optional`, i.e. `stdout is None` would be true. `cast()` is effectively masking this fact, pretending `stdout` is *always* not `None`. `output = self.run(...)` followed by `assert out.stdout is not None` should be enough to "prove" we can safely return `output.stdout`, with no `cast()` involved. Types match, it's the optionality that's preventing simple `return self.run(...).stdout`.